is_iam_enabled.go

```diff
@@ -0,0 +1,76 @@
+package alks
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+)
+
+type IsIamRoleRequest struct {
+	RoleArn string `json:"roleArn"`
```

Comment:
The comment explaining this is further down, but `omitempty` tells the serialization runtime to ignore this value if it's empty-like (empty list, `nil`, empty string, etc.).  Should simplify the code you have below to handle different call types.
```suggestion
	RoleArn string `json:"roleArn,omitempty"`
```